logprep/processor/timestamp_differ/rule.py

```diff
@@ -0,0 +1,97 @@
+"""
+.. _timestamp_differ_rule:
+
+TimestampDiffer
+===============
+
+The `timestamp_differ` processor allows to calculate the time difference between two timestamps.
+The timestamp format can be specified per timestamp. Following patterns can be used to define the
+timestamp format:
+`Timestamp tokens <https://arrow.readthedocs.io/en/latest/guide.html#supported-tokens>`_.
+
+A speaking example:
+
+..  code-block:: yaml
+    :linenos:
+    :caption: Given timestamp differ rule
+
+    filter: 'ingest AND processed'
+    timestamp_differ:
+      diff: ${processed:YYYY-MM-DD HH:mm:ss} - ${ingest:YYYY-MM-DD HH:mm:ss}
+      target_field: processing_time
+      output_format: seconds
+    description: '...'
+
+..  code-block:: json
+    :linenos:
+    :caption: Incoming event
+
+    {"ingest": "2022-12-06 10:00:00", "processed": "2022-12-06 10:00:05"}
+
+..  code-block:: json
+    :linenos:
+    :caption: Processed event
+
+    {"ingest": "2022-12-06 10:00:00", "processed": "2022-12-06 10:00:05", "processing_time": "5 s"}
+
+Examples for timestamp_differ:
+------------------------------
+
+.. datatemplate:import-module:: tests.unit.processor.timestamp_differ.test_timestamp_differ
+   :template: testcase-renderer.tmpl
+"""
+import re
+
+from attr import field
+from attrs import define, validators
+
+from logprep.processor.field_manager.rule import FieldManagerRule
+
+FIELD_PATTERN = r"\$\{([+&?]?[^${]*)\}"
+DEFAULT_TIMESTAMP_PATTERN = "YYYY-MM-DDTHH:mm:ssZZ"
+
+
+class TimestampDifferRule(FieldManagerRule):
+    """TimestampDifferRule"""
+
+    @define(kw_only=True)
+    class Config(FieldManagerRule.Config):
+        """Config for TimestampDiffer"""
+
+        diff: str = field(
+            validator=[
+                validators.instance_of(str),
+                validators.matches_re(rf"({FIELD_PATTERN} - {FIELD_PATTERN})"),
+            ]
+        )
+        """Specifies the timestamp subtraction and their respective timestamp formats. The fields
+        and the timestamp format can be specified in the form of:
+        :code:`${dotted.field.path:timestamp-format}`. If no timestamp format is given, e.g.
+        :code:`${dotted.field.path}`, then the default pattern :code:`"YYYY-MM-DDTHH:mm:ssZZ"` will
+        be used."""
+        source_fields: list = field(factory=list)
```

Comment:
consider, adding a additional field for the formats. this would lead to not breaking the rules with the source fields.

Reply:
Also done